README.md

```diff
@@ -1,2 +1,127 @@
-# python-lab
-Lab activities for general Python programming workshop
+# Python Lab
+
+Lab activities for general Python programming workshop.
+
+Attached is a [`calc.py`](./calc.py) script, which is a basic functional calculator program.
+
+## Tasks
+
+Whilst you can complete the tasks in any order, they start easier and get harder, so we'd recommend completing them in order.
+
+Each task is presented with an example output. This is merely an example - feel free to write it however you wish.
+
+### Show a message when the user enters an invalid operation
+
+```
+Select operation.
+1.Add
+2.Subtract
+3.Multiply
+4.Divide
+Enter choice(1/2/3/4): 5
+Invalid choice - try again
+Enter choice(1/2/3/4): 5
+```
+
+### Rather than entering numbers for each of the operations, use the mathematical symbol ("+" for addition etc)
+
+```
+Select operation:
+- Add (+)
+- Subtract (-)
+- Multiply (*)
+- Divide (/)
+Enter choice: +
+```
+
+### Add support for operating on decimal values
+
+```
+Select operation:
+- Add (+)
+- Subtract (-)
+- Multiply (*)
+- Divide (/)
+Enter choice: +
+Enter first number: 2.5
+Enter second number: 3.5
+2.5 + 3.5 = 6
+```
+
+### When prompted for a number, if someone enters something which isn't a number, show an error message and ask for the value again
```

Comment:
Not sure if the correct version of calc.py is provided as I think the script already does this?